Avatar Changes

emplums opened the pull request:
This PR builds upon #765.

## Changes I've made in this PR:
- Makes `shape`a boolean called `square` (change from the PR this PR is built from)
- Makes the circle avatar the default
- Updated the `square` border radii to `4px` and `6px` instead of `2px` and `3px`
- Updates AvatarPair to be work well with circle avatars, square avatars, and a mix of the two
- Updates AvatarStack to work well with circle avatars, using the new design @simurai came up with and @colinkeany implemented in https://github.com/primer/css/pull/1076
- Updated Avatar docs
- Updated types, added missing `AvatarPair` type
- Removed `isChild` remnants, this prop isn't actually used anywhere anymore
- Gave `Avatar` `COMMON` props instead of just `space` for consistency with other components
- I removed `alt` and `src` from the prop types since they are native html attributes, but left them in the docs since it's important that people use them. This makes `alt` no longer required, but I think that's fine since we set a default for `alt` if the user doesn't provide it to help with a11y best practices of setting alt to a empty string if we want the SR to skip it.
- General clean up, there were a lot of unnecessary duplicate styles in places that I removed

🚨 Breaking Changes 🚨 
- AvatarStack now expects you to use `Avatar` components as children

emplums commented:
Note: I used class names similar to those used in Primer CSS to get the styling right in this one, since there's lots of nesting and complicated styles I felt it was the best way to approach this, but it made me realize that if this component is used anywhere where Primer CSS is loaded globally these components could get messed up if the Primer Components & Primer CSS classes don't have exactly the same rules applied, which is definitely possible because sometimes we clean up CSS when porting it over to Primer Components 😬

I wonder if there's a way to create unique class names for hard-coded class names within components so that we can avoid this?

BinaryMuse commented on docs/content/Avatar.mdx:
```diff
@@ -3,17 +3,23 @@ title: Avatar
 
 
-Avatars are images used to represent users and organizations on GitHub. They typically are squares with rounded edges.
+Avatars are images used to represent users and organizations on GitHub. Use the default round avatar for users, and the `square` prop for organization or any other non-human avatars.
```
Minor typo:

```suggestion
Avatars are images used to represent users and organizations on GitHub. Use the default round avatar for users, and the `square` prop for organizations or any other non-human avatars.
```

BinaryMuse commented on src/AvatarStack.js:
```diff
@@ -1,147 +1,149 @@
 `
 const AvatarStack = ({children = [], alignRight, ...rest}) => {
+  const count = children.length
+  const wrapperClassNames = classnames(
+    count === 2 ? 'pc-AvatarStack--two' : count > 2 ? 'pc-AvatarStack--three-plus' : '',
+    alignRight ? 'pc-AvatarStack--right' : ''
+  )
```
This nested ternary is pretty funky; `classnames` supports a syntax that will allow us to unwrap it:

```suggestion
  const wrapperClassNames = classnames({
    'pc-AvatarStack--two': count === 2,
    'pc-AvatarStack--three-plus': count > 2,
    'pc-AvatarStack--right': alignRight
  })
```

emplums replied:
whaaaa this is awesome!

BinaryMuse commented on src/AvatarStack.js:
```diff
@@ -1,147 +1,149 @@
-        {React.cloneElement(child, {className: 'AvatarItem'})}
-      </>
-    )
+    return <>{React.cloneElement(child, {className: 'pc-AvatarItem', sx: {zIndex: 10 - index}})}</>
```
Two things:

1. If the child had any `className` or `sx` props set, this will overwrite them.
2. I don't think this has to be in a fragment, cause it's only one item:

```suggestion
    return React.cloneElement(child, {className: 'pc-AvatarItem', sx: {zIndex: 10 - index}})
```

emplums commented:
>On hover, the first avatar in the stack looks like it's moving over slightly with the rest of the stack.
Fixed! 
> All hidden avatars should show on hover
Fixed!

> Missing the tooltip on hover that shows the usernames associated with the avatars
This can be done by wrapping the component with a `Tooltip` component - though I just realized there might be some bugs in `Tooltip`, I'll make a separate issue for that :)

emplums commented:
@colinkeany thanks for your work in Primer CSS 💪 that was the base of most of this CSS! 🙌
